### DISCUSSION TITLE
Python 3.10; AssertionError: `Cookie` default value cannot be set in `Annotated` for 'last_query'. Set the default value with `=` instead.

### DISCUSSION THREAD

**Discussion by meowfishorg** [fastapi/fastapi#9309] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import Cookie, Depends, FastAPI

app = FastAPI()


def query_extractor(q: str | None = None):
    return q


def query_or_cookie_extractor(
    q: Annotated[str, Depends(query_extractor)],
    last_query: Annotated[str | None, Cookie()] = None, # Exception if "= None" is added!
):
    if not q:
        return last_query
    return q


@app.get("/items/")
async def read_query(
    query_or_default: Annotated[str, Depends(query_or_cookie_extractor)]
):
    return {"q_or_cookie": query_or_default}
```

### Description

Following the documentation with Python 3.10.10: https://fastapi.tiangolo.com/tutorial/dependencies/sub-dependencies/
If I set the the default of "last_query" to None as in the documentation, an error will occur:
```
AssertionError: `Cookie` default value cannot be set in `Annotated` for 'last_query'. Set the default value with `=` instead.
```

```git
 def query_or_cookie_extractor(
     q: Annotated[str, Depends(query_extractor)],
-    last_query: Annotated[str | None, Cookie()],
+    last_query: Annotated[str | None, Cookie()] = None,
 ):
     if not q:
         return last_query
```
I expect it to work without errors.

### Operating System

Linux

### Operating System Details

Linux Mint 21.1 Cinnamon (5.15.0-67-generic)

### FastAPI Version

0.95.0

### Python Version

3.10.10

**Comment by sharonyogev** [comment, marked as answer]
I think this should solve it: https://github.com/tiangolo/fastapi/pull/9315

**Reply by tiangolo** [reply]
This should be fixed in https://github.com/tiangolo/fastapi/pull/9315

Just released in FastAPI 0.95.1 :tada:
